packages/pds/src/actor-store/blob/transactor.ts

```diff
@@ -177,54 +205,84 @@ export class BlobTransactor extends BlobReader {
       .select('blobCid')
       .execute()
 
-    const newBlobCids = writes
-      .map((w) =>
-        w.action === WriteOpAction.Create || w.action === WriteOpAction.Update
-          ? w.blobs
-          : [],
-      )
-      .flat()
-      .map((b) => b.cid.toString())
     const cidsToKeep = [
-      ...newBlobCids,
+      ...writes
+        .filter((w) => isUpdate(w) || isCreate(w))
+        .flatMap((w) => w.blobs.map((b) => b.cid.toString())),
```

Comment:
I found the `newBlobCids` variable to be a useful way to self-document what's going on:
```ts
    const cidsToKeep = [
      ...newBlobCids,
      ...duplicateCids.map((row) => row.blobCid),
    ]
```

"We're keeping new CIDs plus any that had duplicates."